### PR TITLE
Update pyright to 1.1.372

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

**Comment by srittau** [comment]
Closes: #12425

**Review by AlexWaygood — COMMENTED** [review]
Do the pyright complaints go away if we use `**kwargs: Any` instead of `**kwargs: Unused` for `PurePath.__new__`?

**Comment by srittau** [comment]
Probably. Let's try it.

**Comment by AlexWaygood** [comment]
I guess that didn't work :( worth a try, though

**Review by AlexWaygood — APPROVED** [review]

File: stdlib/pathlib.pyi [hunk 1]
```diff
@@ -101,7 +100,10 @@ class PurePosixPath(PurePath): ...
 class PureWindowsPath(PurePath): ...
 
 class Path(PurePath):
-    def __new__(cls, *args: StrPath, **kwargs: Any) -> Self: ...
+    if sys.version_info >= (3, 12):
+        def __new__(cls, *args: StrPath, **kwargs: Unused) -> Self: ...  # pyright: ignore[reportInconsistentConstructor]
+    else:
+        def __new__(cls, *args: StrPath, **kwargs: Unused) -> Self: ...
```
Comment (srittau) [inline stdlib/pathlib.pyi:106]: (The return type seems wrong to me. It should be `WindowsPath` or `PosixPath`, depending on the platform. But this is out of scope of this PR.)

**Comment by srittau** [comment]
Maybe we should just disable `reportInconsistentConstructor` for stubs. Since we're just copying the upstream annotations, this is out if our control.

**Comment by AlexWaygood** [comment]
> Maybe we should just disable `reportInconsistentConstructor` for stubs. Since we're just copying the upstream annotations, this is out if our control.

I think it has caught a few bugs in the past. It doesn't always end up with us just ignoring the error. But I'm open to it if we don't think the tradeoff is worth it.

**Comment by srittau** [comment]
CI failure is unrelated.
